### PR TITLE
Cold staking

### PR DESCRIPTION
**Pull request by aguycalled**
This PR introduces the changes described in https://github.com/NAVCoin/npips/blob/master/npip-0002.mediawiki.

### CODE REVIEW

File: src/qt/getaddresstoreceive.cpp [hunk 1]
```diff
@@ -73,6 +75,17 @@ void getAddressToReceive::getNewAddress()
     showQR();
 }
 
+void getAddressToReceive::getColdStakingAddress()
+{
+    if (!IsCommunityFundEnabled(pindexBestHeader,Params().GetConsensus()))
```
Comment (Bushstar) [inline src/qt/getaddresstoreceive.cpp:80]: Based on the response in the QMessageBox it seems that this conditional should have used IsColdStakingEnabled.

File: src/rpc/misc.cpp [hunk 2]
```diff
@@ -171,9 +188,13 @@ UniValue validateaddress(const UniValue& params, bool fHelp)
             "  \"isvalid\" : true|false,       (boolean) If the address is valid or not. If not, this is the only property returned.\n"
             "  \"address\" : \"navcoinaddress\", (string) The navcoin address validated\n"
             "  \"scriptPubKey\" : \"hex\",       (string) The hex encoded scriptPubKey generated by the address\n"
+            "  \"stakingaddress\" : \"navcoinaddress\", (string) The navcoin staking address part of a cold staking address\n"
+            "  \"spendingaddress\" : \"navcoinaddress\", (string) The navcoin spending address part of a cold staking address\n"
             "  \"ismine\" : true|false,        (boolean) If the address is yours or not\n"
+            "  \"ismine\" : true|false,        (boolean) If the coins from the address are stakable or not\n"
```
Comment (Bushstar) [inline src/rpc/misc.cpp:194]: Should be labelled as isstakable.

Comment (aguycalled) [reply src/rpc/misc.cpp:194]: already fixed in current master

File: src/qt/getaddresstoreceive.cpp (hunk 1 above)
Comment (aguycalled) [reply src/qt/getaddresstoreceive.cpp:80]: https://github.com/navcoin/navcoin-core/pull/733

File: src/script/script.h [hunk 3]
```diff
@@ -671,6 +673,11 @@ class CScript : public CScriptBase
     unsigned int GetSigOpCount(const CScript& scriptSig) const;
 
     bool IsPayToPublicKeyHash() const;
+    bool IsPayToPublicKey() const;
+
+    bool IsColdStaking() const;
+
+    bool IsColdStake() const;
```
Comment (Bushstar) [inline src/script/script.h:680]: Declared but never defined.

Comment (aguycalled) [reply src/script/script.h:680]: https://github.com/navcoin/navcoin-core/pull/734

File: src/rpc/misc.cpp (hunk 2 above)
Comment (aguycalled) [reply src/rpc/misc.cpp:194]: This pull request is already closed and merged. I would suggest to better check against the current state of master.

Comment (Bushstar) [reply src/rpc/misc.cpp:194]: Thought I did, but I was checking master on the source repo of the PR, not navcoin/navcoin-core. Will pay more attention in future.

**Pull request merged**